### PR TITLE
VACMS-2814: Adding new paragraph checklist field config.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #2814 

## Testing done
Visual / behat

## QA steps
- As an admin, go to `/admin/structure/paragraphs_type/checklist/fields` & `field /admin/structure/paragraphs_type/checklist_item/fields`
- Visually verify that fields on form and in node view mode are ordered with intro below header
- Visually verify that header is limited to 70 chars, and intro is limited to 300 chars.

### CODE REVIEW

File: config/sync/core.entity_form_display.paragraph.checklist.default.yml [hunk 1]
```diff
@@ -37,8 +39,27 @@ content:
     settings:
       size: 60
       placeholder: ''
+      maxlength: 70
+      counter_position: after
+      js_prevent_submit: true
+      count_html_characters: true
+      textcount_status_message: 'Maxlength: <span class="maxlength_count">@maxlength</span><br />Used: <span class="current_count">@current_length</span><br />Remaining: <span class="remaining_count">@remaining_count</span>'
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.paragraph.checklist.default.yml:46]:
```suggestion
      textcount_status_message: 'Characters remaining: <span class="remaining_count">@remaining_count</span>'
```

File: config/sync/core.entity_form_display.paragraph.checklist.default.yml [hunk 2]
```diff
@@ -37,8 +39,27 @@ content:
...
+  field_section_intro:
+    weight: 1
+    settings:
+      rows: 5
+      placeholder: ''
+      maxlength: 300
+      counter_position: after
+      js_prevent_submit: true
+      count_html_characters: true
+      textcount_status_message: 'Maxlength: <span class="maxlength_count">@maxlength</span><br />Used: <span class="current_count">@current_length</span><br />Remaining: <span class="remaining_count">@remaining_count</span>'
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.paragraph.checklist.default.yml:60]:
```suggestion
      textcount_status_message: 'Characters remaining: <span class="remaining_count">@remaining_count</span>'
```

File: config/sync/core.entity_form_display.paragraph.checklist_item.default.yml [hunk 3]
```diff
@@ -24,8 +27,27 @@ content:
     settings:
       size: 60
       placeholder: ''
+      maxlength: 70
+      counter_position: after
+      js_prevent_submit: true
+      count_html_characters: true
+      textcount_status_message: 'Maxlength: <span class="maxlength_count">@maxlength</span><br />Used: <span class="current_count">@current_length</span><br />Remaining: <span class="remaining_count">@remaining_count</span>'
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.paragraph.checklist_item.default.yml:34]:
```suggestion
      textcount_status_message: 'Characters remaining: <span class="remaining_count">@remaining_count</span>'
```

File: config/sync/core.entity_form_display.paragraph.checklist_item.default.yml [hunk 4]
```diff
@@ -24,8 +27,27 @@ content:
...
+  field_section_intro:
+    weight: 1
+    settings:
+      rows: 5
+      placeholder: ''
+      maxlength: 300
+      counter_position: after
+      js_prevent_submit: true
+      count_html_characters: true
+      textcount_status_message: 'Maxlength: <span class="maxlength_count">@maxlength</span><br />Used: <span class="current_count">@current_length</span><br />Remaining: <span class="remaining_count">@remaining_count</span>'
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.paragraph.checklist_item.default.yml:48]:
```suggestion
      textcount_status_message: 'Characters remaining: <span class="remaining_count">@remaining_count</span>'
```

**Comment by ethanteague** [comment]
Thanks for doing that, @oksana-c

**Pull request merged**